### DISCUSSION TITLE
how to use useLocalStorage

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#244] [Web] [answered]
Hi,
I implemented useLocalStorage according to this [template](https://gist.github.com/ThomasKirst/dab6da80cb1bfb96814839c9bba8d70d) by Thomas.

After that my app worked as before (and now it doesn't work any longer) but didn't survive the page reload.
<img width="608" alt="Bildschirmfoto 2022-10-11 um 15 49 11" src="https://user-images.githubusercontent.com/109812125/195109131-b7e0ec81-4f60-4242-992e-262587988272.png">

![Bildschirmfoto 2022-10-11 um 15 50 19](https://user-images.githubusercontent.com/109812125/195109432-c0aa3795-845d-4e5b-8757-593b6ba9fd24.png)
It's the same error two times.

[PullRequest](https://github.com/DanielGennat/capstone-project/pull/17)

**Comment by NikRoe** [comment]
Hey Daniel,

your states were drifting apart. After refreshing, your app did not use what was stored in the local storage.
On top of that, you were missing an `&` in your `if...else`.

We collected all changes in this file for you:
[localStorage.txt](https://github.com/neuefische/questions/files/9757530/localStorage.txt)

Hope this helps.

Best,
Niklas

**Reply by DanielGennat** [reply]
It works but I want to keep my `newTimerChain` `useState` for the form.

**Comment by DanielGennat** [comment, marked as answer]
The problem is when reloading the page the `newTimerChain` is set with the `initilState` of `timerChain` before local Storage has been loaded. So we need a `useEffect` to update `newTimerChain` every time `timerChain` is updated:

```
//EditTimerChainForm.js

  const [newTimerChain, setNewTimerChain] = useState(timerChain);

  useEffect(() => {
    setNewTimerChain(timerChain);
  }, [timerChain]);
```
